[ROCm][PD] add moriio kv connector.

inkcherry opened the pull request:
## Purpose

This PR introduces the mori-io KV connector for AMD devices. Built on top of the [MORI](https://github.com/ROCm/mori) project, the mori-io connector supports both PULL and PUSH modes for KV Cache transfer. Key features include:

- Mori backend integration.​
- Mori-related components (buffer merge &session cache management &batch io).​
- PULL mode (Serial interaction of prefill and decode).​
- PUSH mode (Parallel interaction of prefill and decode, with non-blocking layer-wise transfer)
- A unified proxy example for both push and pull logic.
- xPyD
- parallel strategy support

    |prefill|decode|EP|support|
    |-|-|-|-|
    |TP|TP|✔️|✔️|
    |DP|DP|✔️|✔️|
    |TP|DP|❎|❎|

## High Level Design

<img width="2048" height="2268" alt="image" src="https://github.com/user-attachments/assets/ec9e5618-332a-432a-9f06-9cc0b4b9bfd5" />
Thanks to the well-abstracted vLLM connector interface, we were able to integrate the logic into a single file without affecting other or core files. 

Push Mode Implementation Details:

For prefill, we use a dedicated thread to maintain the transfer queue. Layers are enqueued in a layer-wise manner while the forward pass proceeds normally. Once prefill receives the block allocation signal from decode, it asynchronously schedules transfers.


For decode, Requests first send their allocated blocks to the prefill stage and are then placed into a waiting queue. Once the write completion signal is received from the prefill, the request is extracted from the waiting queue and directly scheduled via continuous batching.


This approach introduces no blocking overhead between prefill and decode, and is compatible with both decode graph mode and the chunked prefill feature.

Regarding the transfer process:
1. Utilizing MORI-IO based RDMA transfer to maximize bandwidth utilization and minimize absolute transfer time in current and future AMD devices.
2. Implementing layer-wise push mode to overlap computation and transfer within a single request to reduce TTFT.

TODO:

- [ ] Support for more topologies (e.g., heterogeneous topologies)
- [ ] Chunk-wise prefill transfer
- [ ] Leverage the first token of prefill in push mode
- [ ] Reduce notify overhead

## Test Plan
device: MI300
### Accuracy

- 1P1D TP8
- 3P2D TP8
- 1P1D DP8
- 1P1D TP8(PULL MODE)
- 1P1D TP2(Qwen3-32B, non-MLA)

### Performance
- 1P1D TP8(PULL  MODE) 2k,4k,8k,16k,32k
- 1P1D TP8(PUSH MODE) 2k,4k,8k,16k,32k

### Accuracy launch scripts

**launch proxy** :```python examples/online_serving/disaggregated_serving/moriio_toy_proxy_server.py```
**gsm8k task**:  The GMS8K proxy remains consistent across all tasks. The startup commands for the instance are listed below.

```
#MODEL_PATH=/nfs/data/Qwen3-32B
MODEL_PATH=/mnt/m2m_nobackup/models/deepseek-ai/DeepSeek-V3
lm_eval --model local-completions --model_args tokenizer=${MODEL_PATH},base_url=http://127.0.0.1:10001/v1/completions,num_concurrent=256,max_retries=1,max_gen_toks=2048 --tasks gsm8k --num_fewshot 5 --batch_size auto  --apply_chat_template
```


**1P1D TP8 or 3P2D TP8**
Launch prefill Instance
```
export VLLM_USE_V1=1    
export VLLM_ROCM_USE_AITER=1 
MODEL_PATH=/mnt/m2m_nobackup/models/deepseek-ai/DeepSeek-V3

vllm serve $MODEL_PATH        \
 -tp 8  \
 --port 20005     \
 --block-size 1          \
 --max-num-batched-tokens 4096         \
 --distributed-executor-backend mp         \
 --gpu_memory_utilization 0.85         \
 --max-model-len 4096         \
 --enforce-eager \
 --max_num_seqs 64\
 --no-enable-prefix-caching \
 --kv-transfer-config '{"kv_connector":"MoRIIOConnector","kv_role":"kv_producer","kv_connector_extra_config":{"proxy_ip":"10.xxx.xxx.xx","proxy_ping_port":"36367","http_port":"20005"}}' 
```

Launch decode instance
```
export VLLM_USE_V1=1    
export VLLM_ROCM_USE_AITER=1 
MODEL_PATH=/mnt/m2m_nobackup/models/deepseek-ai/DeepSeek-V3

vllm serve $MODEL_PATH        \
 -tp 8  \
 --port 40005     \
 --block-size 1          \
 --no-enable-prefix-caching \
 --max-num-batched-tokens 4096         \
 --distributed-executor-backend mp         \
 --gpu_memory_utilization 0.85         \
 --max-model-len 8200         \
 --cuda-graph-sizes 1 256   \
 --trust-remote-code \
 --kv-transfer-config '{"kv_connector":"MoRIIOConnector","kv_role":"kv_consumer","kv_connector_extra_config":{"proxy_ip":"10.xxx.xxx.xx","http_port":"40005","proxy_ping_port":"36367"}}' 

```

**1P1D DP8**
Launch prefill Instance

```
export VLLM_ALL2ALL_BACKEND=naive 
export VLLM_USE_V1=1    
export VLLM_ROCM_USE_AITER=1 
MODEL_PATH=/mnt/m2m_nobackup/models/deepseek-ai/DeepSeek-V3
vllm serve $MODEL_PATH        \
 -tp 1  \
 -dp 8     \
 --enable-expert-parallel         \
 --port 20005     \
 --block-size 1          \
 --no-enable-prefix-caching \
 --max-num-batched-tokens 4096         \
 --distributed-executor-backend mp         \
 --gpu_memory_utilization 0.85         \
 --max-model-len 4096         \
 --enforce-eager \
 --max_num_seqs 64\
 --no-enable-prefix-caching \
 --kv-transfer-config '{"kv_connector":"MoRIIOConnector","kv_role":"kv_producer","kv_connector_extra_config":{"proxy_ip":"10.xxx.xxx.xx","proxy_ping_port":"36367","http_port":"20005"}}' \
 ```
Launch decode instance
  ```
export VLLM_ALL2ALL_BACKEND=naive 
export VLLM_USE_V1=1    
export VLLM_ROCM_USE_AITER=1 

MODEL_PATH=/mnt/m2m_nobackup/models/deepseek-ai/DeepSeek-V3
vllm serve $MODEL_PATH        \
 -tp 1  \
 -dp 8     \
 --enable-expert-parallel         \
 --port 40005     \
 --block-size 1          \
 --no-enable-prefix-caching \
 --max-num-batched-tokens 4096         \
 --distributed-executor-backend mp         \
 --gpu_memory_utilization 0.85         \
 --max-model-len 8200         \
 --cuda-graph-sizes 1 256         \
 --trust-remote-code \
  --max_num_seqs 256\
  --kv-transfer-config '{"kv_connector":"MoRIIOConnector","kv_role":"kv_consumer","kv_connector_extra_config":{"proxy_ip":"10.xxx.xxx.xx","http_port":"40005","proxy_ping_port":"36367"}}' 
```

**1P1D TP8(PULL MODE)**
Launch prefill instance
```
export MORIIO_CONNECTOR_READ_MODE=1
export VLLM_USE_V1=1    
export VLLM_ROCM_USE_AITER=1 
MODEL_PATH=/mnt/m2m_nobackup/models/deepseek-ai/DeepSeek-V3
vllm serve $MODEL_PATH        \
 -tp 8  \
 --port 20005     \
 --block-size 1          \
 --max-num-batched-tokens 32768         \
 --distributed-executor-backend mp         \
 --gpu_memory_utilization 0.85         \
 --max-model-len 32768         \
 --enforce-eager \
 --no-enable-prefix-caching \
 --kv-cache-dtype fp8  \
 --max_num_seqs 64 \
 --kv-transfer-config '{"kv_connector":"MoRIIOConnector","kv_role":"kv_producer","kv_connector_extra_config":{"proxy_ip":"10.xxx.xxx.xx","http_port":"20005","proxy_ping_port":"36367"}}' 


```
Launch decode instance

```
export MORIIO_CONNECTOR_READ_MODE=1
export VLLM_USE_V1=1    
export VLLM_ROCM_USE_AITER=1 
MODEL_PATH=/mnt/m2m_nobackup/models/deepseek-ai/DeepSeek-V3
vllm serve $MODEL_PATH        \
 -tp 8  \
 --port 40005     \
 --block-size 1          \
 --no-enable-prefix-caching \
 --max-num-batched-tokens 32768         \
 --distributed-executor-backend mp         \
 --gpu_memory_utilization 0.85         \
  --max-model-len 32768         \
 --trust-remote-code \
 --kv-cache-dtype fp8  \
 --cuda-graph-sizes 1 256         \
 --kv-transfer-config '{"kv_connector":"MoRIIOConnector","kv_role":"kv_consumer","kv_connector_extra_config":{"proxy_ip":"10.xxx.xxx.xx","http_port":"40005","proxy_ping_port":"36367"}}' 
```



**1P1D TP2(Qwen3-32B, non-MLA)**
Launch prefill instance

```
export VLLM_USE_V1=1    
export VLLM_ROCM_USE_AITER=1 
MODEL_PATH=/nfs/data/Qwen3-32B
vllm serve $MODEL_PATH        \
 -tp 2 \
 --port 20005     \
 --block-size 32         \
 --max-num-batched-tokens 4096         \
 --distributed-executor-backend mp         \
 --gpu_memory_utilization 0.8         \
 --max-model-len 4096         \
 --no-enable-prefix-caching \
      --enforce-eager \
 --kv-transfer-config '{"kv_connector":"MoRIIOConnector","kv_role":"kv_producer","kv_connector_extra_config":{"proxy_ip":"10.194.132.10","proxy_port":"30001","http_port":"20005"}}' 

```
Launch decode instance
```
export VLLM_USE_V1=1    
export VLLM_ROCM_USE_AITER=1 
MODEL_PATH=/nfs/data/Qwen3-32B

vllm serve $MODEL_PATH        \
 -tp 2 \
 --port 40005     \
 --block-size 32          \
   --enforce-eager \
 --no-enable-prefix-caching \
 --max-num-batched-tokens 4096         \
 --distributed-executor-backend mp         \
 --gpu_memory_utilization 0.8         \
 --max-model-len 4096         \
 --trust-remote-code \
 --kv-transfer-config '{"kv_connector":"MoRIIOConnector","kv_role":"kv_consumer","kv_connector_extra_config":{"proxy_ip":"10.194.132.10","proxy_port":"30001","http_port":"40005"}}' 

```

### Performance launch scripts
**server**
The performance test uses the same startup scripts as the accuracy test, with the following key parameters standardized for all performance benchmarks to ensure comparability:
```
max_model_len: 32k
max_num_batched_tokens: 32k
kv_cache_dtype: fp8
```
**bench**
```
# ISL=2048
# ISL=4096
# ISL=8192
# ISL=16384
ISL=32760
OSL=5 
RATIO=0
PORT=10001
CONCURRENCY=1 
MODEL_PATH=/mnt/m2m_nobackup/models/deepseek-ai/DeepSeek-V3
PROMPTS=1
      vllm bench serve  \
        --dataset-name random \
        --model  $MODEL_PATH \
        --random-input-len $ISL \
        --random-output-len $OSL \
        --tokenizer $MODEL_PATH \
        --num-prompt $PROMPTS \
        --random-range-ratio $RATIO \
        --base-url "http://127.0.0.1:$PORT" \
        --backend vllm \
        --max-concurrency $CONCURRENCY \
```



## Test Result
**Accuracy**:
1P1D TP8
 |Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9424|±  |0.0064|
|     |       |strict-match    |     5|exact_match|↑  |0.8082|±  |0.0108|

3P2D TP8
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9439|±  |0.0063|
|     |       |strict-match    |     5|exact_match|↑  |0.8127|±  |0.0107|

1P1D DP8
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9447|±  |0.0063|
|     |       |strict-match    |     5|exact_match|↑  |0.8105|±  |0.0108|


1P1D TP8(PULL MODE, fp8 kv cache dtype)
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9310|±  |0.0070|
|     |       |strict-match    |     5|exact_match|↑  |0.8635|±  |0.0095|

1P1D TP2(Qwen3-32B, non-MLA)
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9439|±  |0.0063|
|     |       |strict-match    |     5|exact_match|↑  |0.9325|±  |0.0069|



**Performance**:
1P1D TP8(PULL/PUSH MODE)
TTFT
| Input Length  | PULL  MODE| PUSH MODE   |
|---------|---------|---------|
| 2k      | 207ms   | 194ms   |
| 4k      | 340ms   | 311ms   |
| 8k      | 584ms   | 536ms   |
| 16k     | 1184ms  | 1090ms  |
| 32k     | 2758ms  | 2614ms  |
---


## Prerequisites 
- Install MORI
mori-io and mori-ep have the same dependencies, mori-ep: https://github.com/vllm-project/vllm/pull/28664

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>
cc   @maning00 @TianDi101 @carlushuang @Duyi-Wang 

---

> [!NOTE]
> <sup>[Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) is generating a summary for commit 7164614fb799c3980a2fa055a6dbf14447202965. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> <sup>[Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) is generating a summary for commit 7e1d978124da2b1662f1007a2aa784d4b93bf542. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Enables RDMA-based disaggregated KV transfer on ROCm via MORI.
> 
> - Adds `MoRIIOConnector` (scheduler/worker) under `vllm/distributed/kv_transfer/kv_connector/v1/moriio/*` with ZMQ handshakes, notify flow, READ/PUSH modes, TP/DP support, and session/offset management
> - Registers connector in `kv_connector/factory.py` and adds env toggles (`VLLM_MORIIO_*`) in `vllm/envs.py`
> - Provides a toy proxy server `examples/online_serving/disaggregated_serving/moriio_toy_proxy_server.py` for prefill/decode service discovery and request routing
> - Updates ROCm install docs to include optional MORI setup and adjusts build steps
> - Extends `docker/Dockerfile.rocm_base` to build/install MORI (new args/envs, deps) and bundle its wheel
> - Adds unit tests for handshake, cache registration, and metadata flow in `tests/v1/kv_connector/unit/test_moriio_connector.py`
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 7e1d978124da2b1662f1007a2aa784d4b93bf542. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

inkcherry committed:
initial commit

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
update gitignore

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
fix with new main branch

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
fix dp router

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
format

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
add proxy example

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
refine

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
fix dp proxy

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
refine code

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
refine

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
refine

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
format

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
format

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
remove port

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
format

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
refine

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
fix format error

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
update

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
fix mypy

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
fix mypy and tp test pass

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
fix all mypy

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
break long line

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
fix format

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
fix all commit

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
refine

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
update

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
update

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
update

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
refine code

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
remove handle_proxy_request

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
more

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
update

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
tp write single pass

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
updata finished request collection

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
format

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
update lock

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
update proxy path

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
format

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
update

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
format

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
Merge branch 'main' into upstream_mori_

inkcherry committed:
update

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

tjtanaa commented on vllm/distributed/kv_transfer/kv_connector/v1/moriio_connector.py:
```diff
@@ -0,0 +1,2305 @@
+        )
+        self.backend_name = backend.get_name()
+        attn_backend = AttentionBackendEnum[self.backend_name]
+        self._use_flashinfer = attn_backend == AttentionBackendEnum.FLASHINFER
```
NITS: Will this be used? can we clean up unused code path?

tjtanaa replied:
Have you by chance tested MORI on non-MLA models? @inkcherry

tjtanaa commented on vllm/distributed/kv_transfer/kv_connector/v1/moriio_connector.py:
```diff
@@ -0,0 +1,2305 @@
+        Args:
+            worker: Reference to the parent worker
+        """
+        # self.worker = worker
```
NITS: can remove comment?

inkcherry committed:
fix format

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
clean up code

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
use envs

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
improve shutdown

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
format

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
clean up

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
Fix the issue of num_block inconsistency in non-MLA scenarios

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
format

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
clean up

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry commented on vllm/distributed/kv_transfer/kv_connector/v1/moriio_connector.py:
```diff
@@ -0,0 +1,2305 @@
+        self.backend_name = backend.get_name()
+        attn_backend = AttentionBackendEnum[self.backend_name]
+        self._use_flashinfer = attn_backend == AttentionBackendEnum.FLASHINFER
+        self._use_pallas = attn_backend == AttentionBackendEnum.PALLAS
```
removed

ChuanLi1101 commented on vllm/distributed/kv_transfer/kv_connector/v1/moriio_connector.py:
```diff
@@ -0,0 +1,2343 @@
+    ) -> tuple[list[int], list[int], list[int]]:
+        n = len(offsets_local)
+        if n == 0:
+            return [], [], []
```
Consider caching merged results per request to avoid recomputation?

ChuanLi1101 reviewed (approved):
Minor issues from my side. Approval from my side.

KuntaiDu commented on vllm/distributed/kv_transfer/kv_connector/v1/moriio_connector.py:
```diff
@@ -0,0 +1,2343 @@
+# SPDX-License-Identifier: Apache-2.0
```
This file is tooooo long, can we separate it into multiple files? As long as you pack all the files inside a folder under `kv_connector/v1/` (except for the connector file itself), feel free to create files.

inkcherry committed:
remove merge

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
fix comments

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
split large file

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
fix type checking

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
add basic test

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
format ut

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
refine ut

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
fix typo

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
Merge remote-tracking branch 'off/main' into mine

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
format

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
ci

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
doc

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
fix main branch

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry commented:
> @inkcherry Please add unit test to https://github.com/vllm-project/vllm/tree/main/tests/v1/kv_connector/unit

hi @tjtanaa ,I've added the requested unit tests and addressed all comments in this PR. Tests are passing as shown below:

test result:
<img width="1409" height="63" alt="image" src="https://github.com/user-attachments/assets/f6a7c440-ae08-4a99-8635-d9fb54eb2787" />


I discussed with @alexsun07, [MoRIEP](https://github.com/vllm-project/vllm/pull/28664) encountered some issues when merging the latest main branch. He'll handle that separately later，therefore, we are placing the Dockerfile changes in this PR first.

Could you take a look this when you have a moment? thanks!

inkcherry committed:
Merge branch 'main' into upstream_mori_

inkcherry committed:
Merge branch 'main' into upstream_mori_

inkcherry committed:
Merge branch 'main' into upstream_mori_

inkcherry committed:
Merge branch 'main' into upstream_mori_

tjtanaa commented:
Add @gshtras in this thread loop as there is an update on the `Dockerfile.rocm_base` and this will need to propagate to the `rocm/vllm-dev:base` later. Moreover, we need to propagate the `rocm/vllm-dev:base` to test this PR with its new unit tests.

tjtanaa reviewed (approved):
LGTM.

kouroshHakha reviewed (commented):
some more qs:

inkcherry commented on examples/online_serving/disaggregated_serving/moriio_toy_proxy_server.py:
```diff
@@ -0,0 +1,320 @@
+        )
+        req_data_to_prefill["kv_transfer_params"]["remote_tp_size"] = (
+            decode_instance_endpoint["tp_size"]
+        )
```
This is due to the RDMA-based push mode. Specifically, D needs to inform P about block information.
If using pull (read) mode, this is not required.

inkcherry replied:
This is due to the RDMA-based push(write) mode. 
If using pull (read) mode, this is not required.

kouroshHakha replied:
can you also provide a proxy example for the pull mode?

inkcherry replied:
Both pull mode and push mode use a unified proxy. For pull mode, you can refer to ```1P1D TP8 (PULL MODE)``` in the test plan of this pr.

inkcherry committed:
refine proxy args

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
Merge remote-tracking branch 'off/main' into tmp_0104

Signed-off-by: inkcherry <mingzhi.liu@amd.com>

inkcherry committed:
Merge branch 'main' into upstream_mori_

inkcherry committed:
Merge branch 'main' into upstream_mori_

tjtanaa committed:
Merge branch 'main' into upstream_mori_

inkcherry committed:
Merge branch 'main' into upstream_mori_

tjtanaa committed:
Merge branch 'main' into upstream_mori_

inkcherry commented on docker/Dockerfile.rocm_base:
```diff
@@ -44,7 +47,7 @@ ENV DEBIAN_FRONTEND=noninteractive
 # Install Python and other dependencies
 RUN apt-get update -y \
-    && apt-get install -y software-properties-common git curl sudo vim less libgfortran5 \
+    && apt-get install -y software-properties-common git curl sudo vim less libgfortran5 libopenmpi-dev libpci-dev \
```
runtime
